src/mongocxx/CMakeLists.txt

```diff
@@ -69,7 +66,6 @@ else()
       find_package(libmongoc-${LIBMONGOC_REQUIRED_ABI_VERSION} ${LIBMONGOC_REQUIRED_VERSION} REQUIRED)
       message ("found libmongoc version ${MONGOC_VERSION}")
       set(libmongoc_target ${MONGOC_LIBRARIES})
-      set(libmongoc_include_directories ${MONGOC_INCLUDE_DIRS})
```

Comment:
Is `libmongoc_include_directories ` not needed? `libmongoc_include_directories` is referenced in MongocxxUtil.cmake.

Reply:
This was removed by the following suggestion. @vector-of-bool would you be able to chime in on the reason for requesting this change? 
![image](https://github.com/mongodb/mongo-cxx-driver/assets/13429210/f3768671-e820-4bd3-ac79-b104ae9085e0)

Reply:
If you link to the libmongoc imported targets, the headers search paths for libmongoc will be propagated into users automatically.